File: orderservice/src/main/java/com/group3/orderservice/service/abstraction/OrderService.java
package com.group3.orderservice.service.abstraction;


import com.group3.orderservice.service.request.PlaceOrderRequest;
import com.group3.orderservice.service.response.GetOrderByIdResponse;

import java.util.List;

public interface OrderService {
    String getUserIdByOrderId(String orderId);

    String placeOrder(PlaceOrderRequest request);

    List<GetOrderByIdResponse> findOrdersByUserId(String userId);

    GetOrderByIdResponse getOrderById(String orderId);
    String getStatus(String orderId);


}

File: frontend/src/api/orderService.js
import axios from 'axios';

const API_URL = 'http://localhost:8081/api/orders';

export const placeOrder = (order) => {
    return axios.post(`${API_URL}/placeOrder`, order);
};
//a
export const getOrderById = (orderId) => {
    return axios.get(`${API_URL}/${orderId}`);
};

export const findOrdersByUserId = (userId) => {
    return axios.get(`${API_URL}/getOrdersByUserId/${userId}`);
};

export const getStatus = (orderId) => {
    return axios.get(`${API_URL}/status/${orderId}`);
};
